### PR TITLE
Add availability information when the symbol does not has its own availability

### PR DESCRIPTION
**Pull request by sofiaromorales**
Bug/issue #, if applicable: 137708623

## Summary

For Objective-C symbols with no explicit availability, some custom logic was preventing the addition of missing introduced versions, leading to issues such as incorrectly adding a platform name to a symbol when it was not available on that platform, and removing fallback platforms.

## Dependencies

_Describe any dependencies this PR might have, such as an associated branch in another repository._

## Testing

Run added unit test with and without the change.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [X] Added tests
- [X] Ran the `./bin/test` script and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

File: Tests/SwiftDocCTests/Infrastructure/SymbolGraph/SymbolGraphLoaderTests.swift [hunk 1]
```diff
@@ -1652,6 +1652,136 @@ class SymbolGraphLoaderTests: XCTestCase {
         XCTAssertEqual(availability.first(where: { $0.domain?.rawValue == "macCatalyst" })?.introducedVersion, SymbolGraph.SemanticVersion(major: 1, minor: 0, patch: 0))
     }
     
+    
+    func testDefaulAvailabilityWhenSymbolIsNotAvailableForThatPlatform() throws {
```
Comment (patshaughnessy) [inline Tests/SwiftDocCTests/Infrastructure/SymbolGraph/SymbolGraphLoaderTests.swift:1656]: Typo `testDefaultAvailabilityWhenSymbolIsNotAvailableForThatPlatform`

File: Tests/SwiftDocCTests/Infrastructure/SymbolGraph/SymbolGraphLoaderTests.swift [hunk 2]
```diff
@@ -1652,6 +1652,136 @@ class SymbolGraphLoaderTests: XCTestCase {
...
+        let infoPlistURL = targetURL.appendingPathComponent("Info.plist")
+        try infoPlist.write(to: infoPlistURL, atomically: true, encoding: .utf8)
+        // Load the bundle & reference resolve symbol graph docs
+        var (_, _, context) = try loadBundle(from: targetURL)
+        guard let availability = (context.documentationCache["c:@F@Bar"]?.semantic as? Symbol)?.availability?.availability else {
+            XCTFail("Did not find availability for symbol 'c:@F@Bar'")
+            return
+        }
+        // Verify we dont add platforms to symbols that are not in that platform SGF. Even if the
+        // platform is part of the default availabilit.
```
Comment (patshaughnessy) [inline Tests/SwiftDocCTests/Infrastructure/SymbolGraph/SymbolGraphLoaderTests.swift:1778]: typo "...of the default availability"

**Review by patshaughnessy — APPROVED** [review]
Nice simplification 👍

**Review by d-ronnqvist — APPROVED** [review]

**Comment by sofiaromorales** [comment]
@swift-ci please test

**Pull request merged**